src/ui/src/writerTypes.ts

```diff
@@ -239,3 +239,28 @@ export type WriterModel = {
 	id: string;
 	name: string;
 };
+
+type CreateAction = {
+	type: "create";
+	components: Component[];
+};
+
+export type LinkActionInfo = {
+	id: ComponentId;
+	newOut: {
+		outId: string;
+		toNodeId: string;
+	};
+};
+
+type LinkAction = {
+	type: "link";
+	links: LinkActionInfo[];
+};
+
+export type AutogenAction = CreateAction | LinkAction;
+
+export type AutogenResult = {
+	actions: (CreateAction | LinkAction)[];
+	messages: string[];
+};
```

Comment:
Please prefix by `Autogen` here, the name is too generic.

I prefer to nest the name to make it more clear
```suggestion
type AutogenActionCreate = {
	type: "create";
	components: Component[];
};
type AutogenActionLinkInfo = {
	id: ComponentId;
	newOut: {
		outId: string;
		toNodeId: string;
	};
};

type AutogenActionLink = {
	type: "link";
	links: AutogenActionLinkInfo[];
};

export type AutogenAction = AutogenActionCreate | AutogenActionLink;

export type AutogenResult = {
	actions: AutogenAction[];
	messages: string[];
};
```